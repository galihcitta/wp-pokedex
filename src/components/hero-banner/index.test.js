import React from "react";
import HeroBanner from "./index";

describe("Hero Banner Component", () => {
    it("renders and create snapshot <HeroBanner />", () => {
        expect(
            toJson(<HeroBanner total={120} unique={20} />),
        ).toMatchSnapshot();
    });

    it("renders success <HeroBanner />", () => {
        expect(shallow(<HeroBanner total={120} unique={20} />).exists()).toBe(
            true,
        );
    });
    it("renders and check data total pokemon <HeroBanner />", () => {
        expect(
            shallow(<HeroBanner total={120} unique={20} />)
                .find({ "test-id": "total-pokemon" })
                .find("span")
                .text(),
        ).toEqual("120");
    });

    it("renders and check data total unique pokemon <HeroBanner />", () => {
        expect(
            shallow(<HeroBanner total={120} unique={20} />)
                .find({ "test-id": "total-unique" })
                .find("span")
                .text(),
        ).toEqual("20");
    });
});
