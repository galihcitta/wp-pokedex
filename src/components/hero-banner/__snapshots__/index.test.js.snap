// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Banner Component renders and create snapshot <HeroBanner /> 1`] = `undefined`;
